Спринт №17

Shep0rt opened the pull request:
- Рефакторинг по полученным рекомендациям из прошлого спринта
- Рефакторинг, DI переход на Koin вместо Creator

Shep0rt committed:
Спринт №17

- Рефакторинг по полученным рекомендациям из прошлого спринта
- Рефакторинг, переход на Koin вместо Creator

IliaAnK commented on app/src/main/java/com/practicum/playlistmaker/presentation/player/PlayerViewModel.kt:
```diff
@@ -7,10 +7,11 @@ import androidx.lifecycle.LiveData
-class PlayerViewModel(private val url: String) : ViewModel() {
+class PlayerViewModel(private val track: TrackUiDto) : ViewModel() {
 
     private val playerStateLiveData = MutableLiveData<PlayerState>(PlayerState.Default)
```
Обязательно:
Не должно быть несколько разных `LiveData`. Из значений полей `playerStateLiveData` и `progressTimeLivaData` нужно сформировать сущность которая будет единственным стейтом экрана.

Shep0rt committed:
Спринт №17

- Поправил SettingsViewModel.kt теперь переключение тем работает через Live Data
- Поправил работу с плеером теперь PlayerViewModel.kt использует единую сущность для работы с LiveData
